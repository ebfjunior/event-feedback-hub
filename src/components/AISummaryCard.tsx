import { Card } from '@/components/ui/card';
import { Badge } from '@/components/ui/badge';
import { Check, AlertTriangle, Bot } from 'lucide-react';

export type AISummaryData = {
  positive_percentage: number;
  top_highlights: string[];
  areas_for_improvement: string[];
} | null;

export function AISummaryCard({ summary }: { summary: AISummaryData }) {
  return (
    <Card className="p-4">
      <div className="mb-3 flex items-center gap-2 text-sm font-medium">
        <span>AI Feedback Summary</span>
        <Badge variant="secondary" className="gap-1">
          <Bot className="h-3.5 w-3.5" /> AI Generated
        </Badge>
      </div>

      {/* Positivity bar */}
      <div className="mb-4">
        <div className="mb-1 text-xs text-muted-foreground">
          {summary ? `${summary.positive_percentage}% Positive` : 'Summary unavailable'}
        </div>
        <div className="h-2 w-full overflow-hidden rounded-full bg-muted">
          <div
            className="h-full rounded-full bg-emerald-500 transition-all"
            style={{ width: `${Math.min(Math.max(summary?.positive_percentage ?? 0, 0), 100)}%` }}
          />
        </div>
      </div>

      {/* Sections */}
      <div className="space-y-4">
        <section className="rounded-md bg-muted/40 p-3">
          <div className="mb-2 flex items-center gap-2 text-sm font-semibold">
            <Check className="h-4 w-4 text-emerald-500" />
            <span>Top Highlights</span>
          </div>
          <ul className="list-inside space-y-2 text-sm text-muted-foreground">
            {(summary?.top_highlights?.length ? summary.top_highlights : ['—']).slice(0, 5).map((item, idx) => (
              <li key={`hi-${idx}`} className="flex gap-2">
                <span className="select-none">•</span>
                <span>{item}</span>
              </li>
            ))}
          </ul>
        </section>

        <section className="rounded-md bg-muted/40 p-3">
          <div className="mb-2 flex items-center gap-2 text-sm font-semibold">
            <AlertTriangle className="h-4 w-4 text-red-500" />
            <span>Areas for Improvement</span>
          </div>
          <ul className="list-inside space-y-2 text-sm text-muted-foreground">
            {(summary?.areas_for_improvement?.length ? summary.areas_for_improvement : ['—']).slice(0, 5).map((item, idx) => (
              <li key={`imp-${idx}`} className="flex gap-2">
                <span className="select-none">•</span>
                <span>{item}</span>
              </li>
            ))}
          </ul>
        </section>
      </div>
    </Card>
  );
}
