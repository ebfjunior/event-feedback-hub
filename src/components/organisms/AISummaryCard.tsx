import { Card } from '@/components/ui/card';
import { Badge } from '@/components/ui/badge';
import { PercentageBar } from '@/components/atoms/PercentageBar';
import { SummarySection } from '@/components/molecules/SummarySection';
import { Check, AlertTriangle, Bot } from 'lucide-react';

export type AISummaryData = {
  positive_percentage: number;
  top_highlights: string[];
  areas_for_improvement: string[];
} | null;

export function AISummaryCard({ summary }: { summary: AISummaryData }) {
  return (
    <Card className="p-4">
      <div className="mb-3 flex items-center gap-2 text-sm font-medium">
        <span>AI Feedback Summary</span>
        <Badge variant="secondary" className="gap-1">
          <Bot className="h-3.5 w-3.5" /> AI Generated
        </Badge>
      </div>

      <div className="mb-4">
        <div className="mb-1 text-xs text-muted-foreground">
          {summary ? `${summary.positive_percentage}% Positive` : 'Summary unavailable'}
        </div>
        <PercentageBar value={summary?.positive_percentage ?? 0} />
      </div>

      <div className="space-y-4">
        <SummarySection icon={<Check className="h-4 w-4 text-emerald-500" />} title="Top Highlights" items={summary?.top_highlights} />
        <SummarySection icon={<AlertTriangle className="h-4 w-4 text-red-500" />} title="Areas for Improvement" items={summary?.areas_for_improvement} />
      </div>
    </Card>
  );
}


