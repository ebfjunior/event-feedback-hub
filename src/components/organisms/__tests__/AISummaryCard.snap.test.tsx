import React from 'react';
import { render } from '@testing-library/react';
import { AISummaryCard } from '../AISummaryCard';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return {
		...actual,
		Check: (p: any) => <svg data-icon="check" {...p} />,
		AlertTriangle: (p: any) => <svg data-icon="alert" {...p} />,
		Bot: (p: any) => <svg data-icon="bot" {...p} />,
	};
});

describe('AISummaryCard (snapshot)', () => {
	it('renders with summary', () => {
		const { container } = render(
			<AISummaryCard
				summary={{ positive_percentage: 75, top_highlights: ['A'], areas_for_improvement: ['B'] }}
			/>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders without summary', () => {
		const { container } = render(<AISummaryCard summary={null} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


