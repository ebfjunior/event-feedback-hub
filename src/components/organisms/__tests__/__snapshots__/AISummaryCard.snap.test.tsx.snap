// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AISummaryCard (snapshot) > renders with summary 1`] = `
<div
  class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border shadow-sm p-4"
  data-slot="card"
>
  <div
    class="mb-3 flex items-center gap-2 text-sm font-medium"
  >
    <span>
      AI Feedback Summary
    </span>
    <span
      class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90 gap-1"
      data-slot="badge"
    >
      <svg
        class="h-3.5 w-3.5"
        data-icon="bot"
      />
       AI Generated
    </span>
  </div>
  <div
    class="mb-4"
  >
    <div
      class="mb-1 text-xs text-muted-foreground"
    >
      75% Positive
    </div>
    <div>
      <div
        class="h-2 w-full overflow-hidden rounded-full bg-muted"
      >
        <div
          class="h-full rounded-full bg-emerald-500 transition-all"
          style="width: 75%;"
        />
      </div>
    </div>
  </div>
  <div
    class="space-y-4"
  >
    <section
      class="rounded-md bg-muted/40 p-3"
    >
      <div
        class="mb-2 flex items-center gap-2 text-sm font-semibold"
      >
        <svg
          class="h-4 w-4 text-emerald-500"
          data-icon="check"
        />
        <span>
          Top Highlights
        </span>
      </div>
      <ul
        class="list-inside space-y-2 text-sm text-muted-foreground"
      >
        <li
          class="flex gap-2"
        >
          <span
            class="select-none"
          >
            •
          </span>
          <span>
            A
          </span>
        </li>
      </ul>
    </section>
    <section
      class="rounded-md bg-muted/40 p-3"
    >
      <div
        class="mb-2 flex items-center gap-2 text-sm font-semibold"
      >
        <svg
          class="h-4 w-4 text-red-500"
          data-icon="alert"
        />
        <span>
          Areas for Improvement
        </span>
      </div>
      <ul
        class="list-inside space-y-2 text-sm text-muted-foreground"
      >
        <li
          class="flex gap-2"
        >
          <span
            class="select-none"
          >
            •
          </span>
          <span>
            B
          </span>
        </li>
      </ul>
    </section>
  </div>
</div>
`;

exports[`AISummaryCard (snapshot) > renders without summary 1`] = `
<div
  class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border shadow-sm p-4"
  data-slot="card"
>
  <div
    class="mb-3 flex items-center gap-2 text-sm font-medium"
  >
    <span>
      AI Feedback Summary
    </span>
    <span
      class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90 gap-1"
      data-slot="badge"
    >
      <svg
        class="h-3.5 w-3.5"
        data-icon="bot"
      />
       AI Generated
    </span>
  </div>
  <div
    class="mb-4"
  >
    <div
      class="mb-1 text-xs text-muted-foreground"
    >
      Summary unavailable
    </div>
    <div>
      <div
        class="h-2 w-full overflow-hidden rounded-full bg-muted"
      >
        <div
          class="h-full rounded-full bg-emerald-500 transition-all"
          style="width: 0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="space-y-4"
  >
    <section
      class="rounded-md bg-muted/40 p-3"
    >
      <div
        class="mb-2 flex items-center gap-2 text-sm font-semibold"
      >
        <svg
          class="h-4 w-4 text-emerald-500"
          data-icon="check"
        />
        <span>
          Top Highlights
        </span>
      </div>
      <ul
        class="list-inside space-y-2 text-sm text-muted-foreground"
      >
        <li
          class="flex gap-2"
        >
          <span
            class="select-none"
          >
            •
          </span>
          <span>
            —
          </span>
        </li>
      </ul>
    </section>
    <section
      class="rounded-md bg-muted/40 p-3"
    >
      <div
        class="mb-2 flex items-center gap-2 text-sm font-semibold"
      >
        <svg
          class="h-4 w-4 text-red-500"
          data-icon="alert"
        />
        <span>
          Areas for Improvement
        </span>
      </div>
      <ul
        class="list-inside space-y-2 text-sm text-muted-foreground"
      >
        <li
          class="flex gap-2"
        >
          <span
            class="select-none"
          >
            •
          </span>
          <span>
            —
          </span>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
